py/kubeflow/testing/run_e2e_workflow.py

```diff
@@ -228,56 +232,104 @@ def run(args, file_handler): # pylint: disable=too-many-statements,too-many-bran
     # are submitting jobs manually for testing/debugging. Since the prow should
     # vend unique build numbers for each job.
     workflow_name += "-{0}".format(salt)
-
     workflow_names.append(workflow_name)
-    # Create a new environment for this run
-    env = workflow_name
 
-    util.run([ks_cmd, "env", "add", env, "--namespace=" + get_namespace(args)],
-              cwd=w.app_dir)
-
-    util.run([ks_cmd, "param", "set", "--env=" + env, w.component,
-              "name", workflow_name],
-             cwd=w.app_dir)
-
-    # Set the prow environment variables.
-    prow_env = []
+    # check if ks workflow and run
+    if hasattr(w, "app_dir"):
+      ks_cmd = ks_util.get_ksonnet_cmd(w.app_dir)
+
+      # Print ksonnet version
+      util.run([ks_cmd, "version"])
+
+      # If we are scoping this workflow to specific directories, check if any files
+      # modified match the specified regex patterns.
+      dir_modified = False
```

Comment:
The only problem is that the Python-defined workflows don't have the `app_dir` property, did you want that property added to them?

Reply:
I think I'm suggesting the code should look something like the following

if skip_workflow
   continue

if is_ksonnet_workflow
   ...
elif is_python_workflow

In other words all of the logic to decide whether the workflow should be skipped or not should be run regardless of what type of workflow is that.

If IIUC we only check app_dir to determine whether it is a ksonnet workflow or not. So that part should come later.

Reply:
See my other comment about lines 195-215

Reply:
Fixed.